src/NUnitFramework/tests/Internal/StringTokenEnumeratorTests.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System.Collections.Generic;
+
+namespace NUnit.Framework.Internal
+{
+    public class StringTokenEnumeratorTests
+    {
+        [Test]
+        public void Empty()
+        {
+            var tokens = Enumerate("", ',');
+            Assert.AreEqual(0, tokens.Count);
+        }
+
+        [Test]
+        public void Whitespace()
+        {
+            var tokens = Enumerate("  ", ',');
+            Assert.AreEqual(1, tokens.Count);
+            Assert.AreEqual("  ", tokens[0]);
+        }
+
+        [Test]
+        public void Tokens()
+        {
+            var tokens = Enumerate("a,b,c", ',');
+            Assert.AreEqual(3, tokens.Count);
+            Assert.AreEqual("a", tokens[0]);
+            Assert.AreEqual("b", tokens[1]);
+            Assert.AreEqual("c", tokens[2]);
+        }
+
```

Comment:
Can we also add some tests where we have longer strings between the tokens and also some where the tokens are next to each other.

EDIT: I made the following tests and one of them fail as the string ends with a token
```csharp
        [TestCase("aaa,,cccdd", ',')]
        [TestCase("somestring;another", ';')]
        [TestCase(",abc,def,,somelongerstring,", ',')] // Fails
        public void CompareAgainstStringSplit(string input, char separator)
        {
            var tokens = Enumerate(input, separator);
            var tokens2 = input.Split(separator);
            Assert.That(tokens, Is.EqualTo(tokens2));
        }
```
The failing testcase fails with 
```
  Expected is <System.String[6]>, actual is <System.Collections.Generic.List`1[System.String]> with 5 elements
  Values differ at index [5]
  Missing:  < <string.Empty> >
```

Reply:
Great find, I've modified the logic to account for this and also support to parametrize whether to return empty tokens. By default empty ones are not returned, adjusted this test case to test both.